ee/desktop/notify/notify.go

```diff
@@ -0,0 +1,52 @@
+package notify
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/ee/desktop/assets"
+)
+
+type DesktopNotifier struct {
+	logger       log.Logger
+	iconFilepath string
+}
+
+type Notification struct {
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+func NewDesktopNotifier(logger log.Logger, iconDir string) *DesktopNotifier {
+	notifier := &DesktopNotifier{
+		logger: logger,
+	}
+
+	iconPath, err := setIconPath(iconDir)
+	if err != nil {
+		level.Error(notifier.logger).Log("msg", "could not set icon path for notifications", "err", err)
+	} else {
+		notifier.iconFilepath = iconPath
+	}
+
+	return notifier
+}
+
+func setIconPath(iconDir string) (string, error) {
+	expectedLocation := filepath.Join(iconDir, assets.KolideIconFilename)
+
+	_, err := os.Stat(expectedLocation)
+
+	if os.IsNotExist(err) {
```

Comment:
I don't this this check is necessary, from the os.WriteFile doc:

> WriteFile writes data to the named file, creating it if necessary.

Reply:
The check is in case we've already written out the icon file to this location before. This maybe made sense before I did the last refactor (because I'd previously put the icon file in the root directory), but if we keep using the temp directory instead (https://github.com/kolide/launcher/pull/985#discussion_r1066399804), then probably it makes sense to remove?

Reply:
ohh, I see now ... the check is preventing re-writing the file, so yea, disregard =)

Reply:
You mean, have the launcher non-desktop side write it (to a non-temp location) and pass that as location in a flag to the user side? I like that and had been thinking about doing it that way, but had been waiting for CJ's phase 3 to land since it does something similar-ish with the menu data file

Reply:
Yes, something like that. I'm thinking about two things.

First, I think the desktop process (Which runs as the user) should avoid creating files. It's harder to manage permissions and error cases. There are times it's going to be unavoidable, but when we can avoid it, we should.

Second, we have these long lived fixed assets. It seems reasonable to have launcher drop them into somewhere for consumption by all the pieces.